react-anime/src/App.jsx

```diff
@@ -0,0 +1,106 @@
+import { useState, useRef } from 'react'
+import {fetchDataAll} from './JS/api'
+import ApiSingleDiv from './components/ApiSingleDiv'
+import ListItem from './components/ListItem'
+import Form from './components/Form'
+
+function App() {
+  const [animes, setAnimes] = useState([])
+  const [watchList, setWatchList] = useState([])
+  const animeSearch = useRef(null);
+  
+  const handleSubmit = async (event) => {
+    event.preventDefault();
+    const search = animeSearch.current.value;
+    
+    const allAnimes = await fetchDataAll(search)
+    const {data} = allAnimes;
+    
+    const animesList = [];
```

Comment:
You do not need to do lines 19-30. This is completely unnecessary and doing extra work for nothing. After line 17 you can set animes to data. You can then reference animes by the obj keys in ApiSingleDiv.